### PR TITLE
Delegate all room alias validation to the RoomAliasField validator

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/11732
Fixes https://github.com/vector-im/riot-web/issues/11737

alias

Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

### CODE REVIEW

**Comment by aaronraimist** [comment]
It probably shouldn't say this alias is available to use when you haven't typed anything in yet

**Comment by t3chguy** [comment]
I don't disagree but this PR does not change the validator:

image

**Comment by t3chguy** [comment]
@aaronraimist seems like this was done on purpose: https://github.com/matrix-org/matrix-react-sdk/blob/ebf7eb698dd548766ec8dfb3f76438dbb13b1f64/src/components/views/elements/RoomAliasField.js#L97

File: src/components/views/room_settings/AliasSettings.js [hunk 1]
```diff
@@ -99,11 +118,6 @@ export default class AliasSettings extends React.Component {
         return dict;
     }
 
-    isAliasValid(alias) {
-        // XXX: FIXME https://github.com/matrix-org/matrix-doc/issues/668
-        return (alias.match(/^#([^/:,]+?):(.+)$/) && encodeURI(alias) === alias);
```
Comment (turt2live) [inline src/components/views/room_settings/AliasSettings.js]: we lost this validation somewhere...

Comment (t3chguy) [reply src/components/views/room_settings/AliasSettings.js]: As per the title of the PR: Delegate all room alias validation to the RoomAliasField validator, i.e: https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/components/views/elements/RoomAliasField.js#L80-L86

Comment (turt2live) [reply src/components/views/room_settings/AliasSettings.js]: /me tries to not flip a table at the duplication

Comment (t3chguy) [reply src/components/views/room_settings/AliasSettings.js]: hence fixing https://github.com/vector-im/riot-web/issues/11737 :P

Comment (t3chguy) [reply src/components/views/room_settings/AliasSettings.js]: hence fixing vector-im/riot-web#11737 :P

**Pull request merged**